server/segment_rpc_test.go

```diff
@@ -469,6 +477,261 @@ func TestServeSegment_InsufficientBalanceError(t *testing.T) {
 	assert.Equal("Internal Server Error", strings.TrimSpace(string(body)))
 }
 
+func TestServeSegment_DebitFees_SingleRendition(t *testing.T) {
+	orch := &mockOrchestrator{}
+	handler := serveSegmentHandler(orch)
+
+	require := require.New(t)
+
+	orch.On("VerifySig", mock.Anything, mock.Anything, mock.Anything).Return(true)
+
+	s := &BroadcastSession{
+		Broadcaster: stubBroadcaster2(),
+		ManifestID:  core.RandomManifestID(),
+		Profiles: []ffmpeg.VideoProfile{
+			ffmpeg.P720p60fps16x9,
+		},
+	}
+	seg := &stream.HLSSegment{Data: []byte("foo")}
+	creds, err := genSegCreds(s, seg)
+	require.Nil(err)
+
+	md, err := verifySegCreds(orch, creds, ethcommon.Address{})
+	require.Nil(err)
+
+	orch.On("ProcessPayment", net.Payment{}, s.ManifestID).Return(nil)
+	orch.On("SufficientBalance", s.ManifestID).Return(true)
+
+	tData := &core.TranscodeData{
+		Data:   []byte("foo"),
+		Pixels: int64(110592000),
+	}
+	tRes := &core.TranscodeResult{
+		TranscodeData: []*core.TranscodeData{tData},
+		Sig:           []byte("foo"),
+		OS:            drivers.NewMemoryDriver(nil).NewSession(""),
+	}
+	orch.On("TranscodeSeg", md, seg).Return(tRes, nil)
+	orch.On("DebitFees", md.ManifestID, mock.Anything, mock.Anything)
+
+	headers := map[string]string{
+		paymentHeader: "",
+		segmentHeader: creds,
+	}
+	resp := httpPostResp(handler, bytes.NewReader(seg.Data), headers)
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	require.Nil(err)
+
+	var tr net.TranscodeResult
+	err = proto.Unmarshal(body, &tr)
+	require.Nil(err)
+
+	assert := assert.New(t)
+	assert.Equal(http.StatusOK, resp.StatusCode)
+
+	res, ok := tr.Result.(*net.TranscodeResult_Data)
+	assert.True(ok)
+	assert.Equal([]byte("foo"), res.Data.Sig)
+	assert.Equal(1, len(res.Data.Segments))
+	assert.Equal(res.Data.Segments[0].Pixels, tData.Pixels)
+	orch.AssertCalled(t, "DebitFees", md.ManifestID, mock.Anything, tData.Pixels)
```

Comment:
I think we can replace `mock.Anything` with the actual `net.PriceInfo` we expect `DebitFees()` to be called with